R/api.R

```diff
@@ -229,3 +230,34 @@ endpoint_available_reports <- function(runner) {
     returning = returning_json("AvailableReports.schema")
   )
 }
+
+target_report_parameters <- function(runner, report_id, commit) {
+  tryCatch(
+    parameters <- runner$get_report_parameters(report_id, commit),
+    error = function(e) {
+      pkgapi::pkgapi_stop(e$message, "FAILED_RETRIEVE_PARAMS")
+    }
+  )
+  if (is.null(names(parameters)) && !is.null(parameters)) {
```

Comment:
Tiny thing but should the order here be reversed?

```
!is.null(parameters) && is.null(names(parameters))
```

as I started getting confused about why reports with no parameters were invalid. Or we could use an early exit:

```
if (is.null(parameters)) {
  return(list())
}
if (!is.null(names(parameters))) {
  # error
}
# rest of logic
```

Reply:
Reversed the order of that, should hopefully be a bit clearer now.